lib/tests/test_revset.rs

```diff
@@ -954,6 +963,14 @@ fn test_evaluate_expression_root_and_checkout() {
         vec![root_commit.id().clone()]
     );
 
+    // but not in the root operation. It might be okay to pretend that the root
+    // commit exists in the root operation, but queries like "root()" shouldn't
+    // panic in any case.
```

Comment:
Yes, seems to me like it the root operation should have the root commit. I guess that would mean that the op store would have to learn about the root commit id, which I don't think it currently knows about.

Reply:
But, to be clear, I'm not asking for any changes to this PR

Reply:
Yes, it's a bit odd that `OpStore` knows about the root commit. If we had an abstraction layer on top of the backend `OpStore`, an empty view might be handled there.